src/horizon_client.rs

```diff
@@ -17,183 +14,434 @@ use reqwest;
+    /// well-formed and appropriate for establishing a connection.
+    /// 
     /// # Arguments
-    /// * `base_url` - The base URL for the Horizon server
+    /// * `base_url` - A `String` representing the base URL of the Horizon server.
+    /// 
     /// # Returns
-    /// The Horizon client
+    /// If successful, this method returns a `Result` containing the initialized `HorizonClient` 
+    /// instance. If the URL validation fails, it returns an error encapsulated within `Result`.
+    /// 
+    /// # Example
+    /// ```rust
+    /// # use stellar_rust_sdk::horizon_client::HorizonClient;
+    /// let horizon_client = HorizonClient::new("https://horizon-testnet.stellar.org".to_string())
+    ///     .expect("Failed to create HorizonClient");
+    /// ```
     pub fn new(base_url: String) -> Result<Self, String> {
         url_validate(&base_url)?;
         Ok(Self { base_url })
     }
 
-    /// Gets an account list from the server
+    /// Retrieves a list of accounts filtered by specific criteria.
+    /// 
+    /// This method retrieves a list of accounts from the Horizon server, filtering the results 
+    /// based on one of four categories: sponsor, signer, asset, or liquidity pool.
+    /// 
+    /// Adheres to the <a href="https://developers.stellar.org/api/horizon/resources/list-all-accounts">List All Accounts</a>
+    /// endpoint.
+    /// 
     /// # Arguments
-    /// * `self` - The Horizon client
-    /// * request - The accounts request
+    /// * `request` - A reference to an implementation of the [`ValidAccountsRequest`] trait, 
+    /// which specifies the filter criteria for the account list request.
+    ///
     /// # Returns
-    /// The accounts response
-    /// # Errors
-    /// Returns an error if the request fails
-    /// [GET /accounts](https://www.stellar.org/developers/horizon/reference/endpoints/accounts.html)
+    /// If successful, this method returns a `Result` containing an [`AccountsResponse`], 
+    /// which encapsulates the list of accounts retrieved from the server. 
+    /// In case of a failure in the request, it returns an error encapsulated within `Result`.
+    ///
+    /// # Example
+    /// To use this method, create an instance of [`AccountsRequest`] and set at least
+    /// one of the four filter options. For example, filtering by signer:
+    /// 
+    /// ```rust
+    /// # use stellar_rust_sdk::accounts::prelude::*;
+    /// # use stellar_rust_sdk::accounts::accounts_request::Signer;
+    /// # use stellar_rust_sdk::models::Request;
+    /// # use stellar_rust_sdk::horizon_client::HorizonClient;
+    /// # 
+    /// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let base_url = "https://horizon-testnet.stellar.org".to_string();
+    /// # let horizon_client = HorizonClient::new(base_url)
+    /// #    .expect("Failed to create Horizon Client");
+    /// let request: AccountsRequest<NoSponsor, Signer, NoAsset, NoLiquidityPool> = AccountsRequest::new()
```

Comment:
I wonder if people would annotate their variable like this. Perhaps to not make it too complicated, just stick with 
```suggestion
    /// let request = AccountsRequest::new()
```